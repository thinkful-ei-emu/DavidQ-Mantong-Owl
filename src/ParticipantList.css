.list{
 width:15%;
 height:100%;
 overflow: scroll;
 display:inline;
}