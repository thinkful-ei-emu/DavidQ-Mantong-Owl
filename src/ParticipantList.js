import React from 'react';
import Participant from './Participant';

export default function ParticipantList(props){
  const people = props.participants.map((person, index)=>{
    return (<Participant key={index} name={person.name} avatar = {person.avatar} inSession = {person.inSession} onStage={person.onStage} />); 
  });

  return(<div>
{people}
  </div>);
} 