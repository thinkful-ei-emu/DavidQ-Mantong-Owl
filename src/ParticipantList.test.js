import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantList from './ParticipantList';
import renderer from 'react-test-renderer';


describe('<ParticipantList />', () => {
    //smoke testing
    it ('render without crashing', ()=> {
        const div =document.createElement('div');
        ReactDOM.render(<ParticipantList participants={test_participants} />,div);

        ReactDOM.unmountComponentAtNode(div);
    });
    //snapshot testing
    it ('renders this UI as expected', ()=>{
        const tree = renderer.create(<ParticipantList participants={test_participants} />).toJSON();
        expect(tree).toMatchSnapshot();     
    });
});
const test_participants = [
  {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  },
  {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
  {
      id: 4,
      name: 'Frank Runciman',
      avatar:
          'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 5,
      name: 'Ashla Attwool',
      avatar:
          'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }
];

