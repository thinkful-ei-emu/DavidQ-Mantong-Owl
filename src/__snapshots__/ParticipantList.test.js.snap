exports[`<ParticipantList /> renders this UI as expected 1`] = `
<div>
  <div>
     
    Koren Templeton
    <img
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1" />
    <p>
      on stage
    </p>
  </div>
  <div>
     
    Caty Flucker
    <img
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1" />
    <p>
      left
    </p>
  </div>
  <div>
     
    Axe Kubicka
    <img
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1" />
    <p>
      left
    </p>
  </div>
  <div>
     
    Frank Runciman
    <img
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1" />
    <p>
      left
    </p>
  </div>
  <div>
     
    Ashla Attwool
    <img
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1" />
    <p>
      on stage
    </p>
  </div>
</div>
`;
